python/python/tests/test_commit_index.py

```diff
@@ -84,3 +92,189 @@ def test_commit_index(dataset_with_index, test_table, tmp_path):
         )
         plan = scanner.explain_plan()
         assert "MaterializeIndex" in plan
+
+
+@pytest.fixture()
+def tmp_tables() -> List[pa.Table]:
+    tables = [
+        {
+            "text": [
+                "Frodo was a puppy",
+                "There were several kittens playing",
+            ],
+            "sentiment": ["neutral", "neutral"],
+        },
+        {
+            "text": [
+                "Frodo was a happy puppy",
+                "Frodo was a very happy puppy",
+            ],
+            "sentiment": ["positive", "positive"],
+        },
+        {
+            "text": [
+                "Frodo was a sad puppy",
+                "Frodo was a very sad puppy",
+            ],
+            "sentiment": ["negative", "negative"],
+        },
+    ]
+    for tb in tables:
+        tb["text2"] = tb["text"]
+        tb["text3"] = tb["text"]
+    return [pa.table(tb) for tb in tables]
+
+
+def test_indexed_unindexed_fragments(tmp_tables, tmp_path):
+    ds = lance.write_dataset(tmp_tables[0], tmp_path, mode="overwrite")
+    ds = lance.write_dataset(tmp_tables[1], tmp_path, mode="append")
+    ds = lance.write_dataset(tmp_tables[2], tmp_path, mode="append")
+    frags = [f for f in ds.get_fragments()]
+    index = ds.create_scalar_index(
+        "text", "INVERTED", fragment_ids=[frags[0].fragment_id]
+    )
+    assert isinstance(index, dict)
+
+    indices = [index]
+    create_index_op = lance.LanceOperation.CreateIndex(
+        new_indices=indices,
+        removed_indices=[],
+    )
+    ds = lance.LanceDataset.commit(
+        ds.uri,
+        create_index_op,
+        read_version=ds.version,
+    )
+
+    unindexed_fragments = ds.unindexed_fragments("text_idx")
+    assert len(unindexed_fragments) == 2
+    assert unindexed_fragments[0].id == frags[1].fragment_id
+    assert unindexed_fragments[1].id == frags[2].fragment_id
+
+    indexed_fragments = [f for fs in ds.indexed_fragments("text_idx") for f in fs]
+    assert len(indexed_fragments) == 1
+    assert indexed_fragments[0].id == frags[0].fragment_id
+
+
+def test_dfs_query_then_fetch(tmp_tables, tmp_path):
+    ds = lance.write_dataset(tmp_tables[0], tmp_path, mode="overwrite")
+    ds = lance.write_dataset(tmp_tables[1], tmp_path, mode="append")
+    ds = lance.write_dataset(tmp_tables[2], tmp_path, mode="append")
+    indices = []
+    frags = list(ds.get_fragments())
+    for f in frags[:2]:
+        # we can create an inverted index distributely
+        index = ds.create_scalar_index("text", "INVERTED", fragment_ids=[f.fragment_id])
```

Comment:
I didn't understand the comment here. Why is this called `distributely `? Here,  a more accurate expression is: Do we provide a mechanism that can be created in parallel or distributed?

Reply:
Got it, maybe a better title for this pr is: Support building invert index in a distributed manner?